pkg/gcrane/gc.go

```diff
@@ -17,62 +17,182 @@ package gcrane
 import (
 	"fmt"
 	"log"
+	"net/http"
+	"time"
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/spf13/cobra"
 )
 
 func init() { Root.AddCommand(NewCmdGc()) }
 
 func NewCmdGc() *cobra.Command {
+	before := ""
+	after := ""
+
 	recursive := false
+	remove := false
+	untag := false
+
 	cmd := &cobra.Command{
 		Use:   "gc",
-		Short: "List images that are not tagged",
+		Short: "Garbage collect images",
 		Args:  cobra.ExactArgs(1),
 		Run: func(_ *cobra.Command, args []string) {
-			gc(args[0], recursive)
+			opts, err := parseOpts(before, after, recursive, remove, untag)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			if err := GarbageCollect(args[0], opts); err != nil {
+				log.Fatalln(err)
+			}
 		},
 	}
 
+	// Filters
+	cmd.Flags().StringVarP(&before, "before", "b", "", "Match images uploaded before this time (RFC3339)")
+	cmd.Flags().StringVarP(&after, "after", "a", "", "Match images uploaded after this time (RFC3339)")
+
+	// Behaviors
 	cmd.Flags().BoolVarP(&recursive, "recursive", "r", false, "Whether to recurse through repos")
+	cmd.Flags().BoolVarP(&remove, "delete", "D", false, "Delete images instead of just printing them")
+	cmd.Flags().BoolVarP(&untag, "untag", "U", false, "Also untag matching images")
 
 	return cmd
 }
 
-func gc(root string, recursive bool) {
+type GCOptions struct {
```

Comment:
No functional options?

When we figure out how to reuse functional opts it would be cool to be able to have callers pass the keychain, transport, etc., too.

Reply:
I'm not totally convinced re: functional options for this, mostly because I'm not sure how to massage the type system into doing what I want without duplicating the world again :(